rubocop/missing_keys_on_shared_area.rb

```diff
@@ -0,0 +1,82 @@
+require 'rubocop'
+
+module RuboCop
+  module Lint
+    class MissingKeysOnSharedArea < RuboCop::Cop::Cop
+      MISSING_KEYS_MSG = "There are missing keys on the shared area. Check constants provided in 'SharedValues' or keys provided in 'output' method in the action's code".freeze
+      MISSING_OUTPUT_METHOD_MSG = "There are declared keys on the shared area 'SharedValues', but 'output' method has not been found".freeze
+
+      attr_writer :shared_values_constants
+      def shared_values_constants
+        @shared_values_constants ||= []
+      end
+
+      def on_module(node)
+        name, body = *node
+        return unless name.source == 'SharedValues'
+        return if body.nil?
+
+        body.children.compact.each do |child|
+          next unless child.respond_to?('casgn_type?')
+
+          if child.or_asgn_type? # Multiple constant values
+            lhs, _value = *child
+            _scope, const_name = *lhs
+          else # Single constant value
+            _scope, const_name, _value = *child
+            self.shared_values_constants << const_name.to_s if const_name
+          end
+
+          _scope, const_name, _value = *body unless const_name
+          self.shared_values_constants << const_name.to_s if const_name
+        end
+      end
+
+      def on_defs(node)
```

Comment:
Similarly you could define a node searcher which gives you the output method directly:

```
def_node_searcher :find_output_method, '(def :output ...)'
```

Reply:
Hey @milch 🙂 

Thanks for your suggestions 🙇 Unfortunately this one does not work. Please note we are trying to detect a static `output` method. I was playing with:

```ruby
def_node_matcher :find_output_method, <<-PATTERN
        (defs (self) :output ...)
      PATTERN
```

Although 

```ruby
def on_class(node)
  output = find_output_method(node)
```

gave me nothing ☝️ 

Any idea? 🤔 🙏

Reply:
I think searching for a node this way is the same as:

```
def on_defs(node) 
  ...
  
  _definee, method_name, _args, body = *node
  return unless method_name.to_s == 'output'
```

We still need to check if the body of the method is not empty and if the body type of the method is an array. 

So I would leave it as it is, WDYT? 🤔